tests/sqlparser_common.rs

```diff
@@ -15357,6 +15357,28 @@ fn check_enforced() {
     );
 }
 
+#[test]
+fn join_precedence() {
+    all_dialects_except(|d| d.is::<SnowflakeDialect>()).verified_query_with_canonical(
```

Comment:
By switching to the dialect method, we would be able to change these to use `all_dialects_where(|d|d.supports_parens_less_nested_join())`